integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,164 @@
+from typing import Any, AsyncGenerator, Optional, List
+from httpx import Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+from port_ocean.utils.cache import cache_iterator_result
+
+TEAM_ID = "__team_id"
+WEBHOOK_EVENTS = [
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+]
+
+
+class ClickupClient:
+    """Clickup client to interact with Clickup API."""
+
+    def __init__(self, clickup_url: str, clickup_token: str):
+        self.clickup_url = clickup_url
+        self.clickup_token = clickup_token
+        self.api_url = f"{self.clickup_url}/api/v2"
+        self.client = http_async_client
+        self.client.timeout = Timeout(30)
+
+    @property
+    def api_headers(self) -> dict[str, Any]:
+        return {
+            "Authorization": self.clickup_token,
+            "Content-Type": "application/json",
+        }
+
+    async def _send_api_request(
+        self,
+        url: str,
+        params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+        method: str = "GET",
+    ) -> Any:
+        """Send a request to the Clickup API."""
+        logger.debug(f"Sending request {method} to endpoint {url}")
+        response = await self.client.request(
+            url=url,
+            method=method,
+            headers=self.api_headers,
+            params=params,
+            json=json_data,
+        )
+        response.raise_for_status()
+        return response.json()
+
+    @cache_iterator_result()
+    async def get_clickup_teams(self) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all Clickup teams."""
+        url = f"{self.api_url}/team"
+        yield (await self._send_api_request(url)).get("teams", [])
```

Comment:
Since this is not doing any pagination, change from  `AsyncGenerator[List[dict[str, Any]], None]` to `List[dict[str, Any]]` and use `return` instead of `yield`. And apply it to all the other functions except the get_paginated_issues()